app/services/tag_adjustment_creation_service.rb

```diff
@@ -20,14 +20,30 @@ def update_tags_and_notify
   private
 
   def update_article
-    article.update!(tag_list: article.tag_list.remove(@tag_adjustment.tag_name)) if @tag_adjustment.adjustment_type == "removal"
+    if @tag_adjustment.adjustment_type == "removal"
+      removed_tags = article.tag_list.select { |tag| tag.casecmp(@tag_adjustment.tag_name).zero? }
+      return if removed_tags.empty?
```

Comment:
Don't even bother touching the polymorphic sorcery of the `acts-as-taggable-on-gem` if we don't have any removed tags